### ISSUE TITLE
Overriding payload in Integration Event (Hubspot)

### ISSUE THREAD

**Issue by webvader** [verbb/formie#713]
**What are you trying to do?**
The `beforeSendPayload` event can be used to stop a `payload` from being sent to an integration. It would be nice if we could alter the `payload` before sending it to the integration. We have a heavy Hubspot flow that uses Hubspot forms to register and add users to workflows within Hubspot. We have to alter some of the initial values that Formie puts in the payload.

**What's your proposed solution?**
I think everything is in place to do this. The function `beforeSendPayload` in `Crm.php` only returns `$event->isValid` it should return the whole `$event`. Then we can do this below line 353 in Hubspot.php (as example):

```
$event = $this->beforeSendPayload($submission, $endpoint, $payload, $method);

if(!$event->isValid) {
    return true;
}

$method = $event->method;
$uri = $event->endpoint;
$options = [
    'json' => $event->payload,
];
```
This makes it very easy to alter `$payload`, `$method` & `$endpoint` before submitting it to the integration.

**Comment by engram-design** [comment]
You should be able to modify the `$endpoint`, `$payload`, and `$method` variables because they're passed to `beforeSendPayload()` by reference, and they are modified in the function from the variables in the event https://github.com/verbb/formie/blob/craft-3/src/base/Crm.php#L125-L127

Is altering `$event->payload` not working for you?

**Comment by webvader** [comment]
Hi Josh,

No, it is not working with an Hubspot integration that uses a Forms endpoint.
I really do not think it can if you look at the Forms integration code:
https://github.com/verbb/formie/blob/551e81e0ed5f739feaee4fc45d7f387374b0f85d/src/integrations/crm/HubSpot.php#L307-L365

That is way I suggested this change. Thanks!

**Comment by engram-design** [comment]
Ah, of course. I forgot that the forms functionality had to be a bit different from the rest of the data objects, so yes that makes sense. Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.5",
  "...": "..."
}
```

Then run `composer update`.

**Comment by webvader** [comment]
Thanks, Josh!

**Comment by engram-design** [comment]
Fixed in [1.5.6](https://github.com/verbb/formie/releases/tag/1.5.6)
